test/harvest/test_dimensions.py

```diff
@@ -234,35 +229,67 @@ def test_fill_in(
     assert "filled in 1 publications" in caplog.text
 
 
-@pytest.fixture
-def mock_dimensions_not_found(monkeypatch):
-    """
-    Mock our function for fetching publications by DOI from Dimensions.
-    """
+def test_fill_in_no_dimensions(
+    snapshot,
+    test_session,
+    mock_publication,
+    mock_no_dim_publication,
+    caplog,
+    monkeypatch,
+):
+    caplog.set_level(logging.INFO)
 
-    def f(*args, **kwargs):
-        raise StopIteration
+    # make it look like Dimensions returns no publications by DOI
```

Comment:
I found moving the monkeypatching into the function to be a bit more readable, than keeping track of the various mocks.